uikit/viewmodels/src/main/java/com/sygic/maps/uikit/viewmodels/common/navigation/preview/RouteDemonstrationManagerImpl.kt

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2019 Sygic a.s. All rights reserved.
+ *
+ * This project is licensed under the MIT License.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.uikit.viewmodels.common.navigation.preview
+
+import androidx.annotation.RestrictTo
+import androidx.lifecycle.MutableLiveData
+import com.sygic.maps.uikit.viewmodels.common.navigation.preview.state.DemonstrationState
+import com.sygic.sdk.route.RouteInfo
+import com.sygic.sdk.route.simulator.RouteDemonstrateSimulator
+
+private const val DEFAULT_SPEED = 1F
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+object RouteDemonstrationManagerImpl : RouteDemonstrationManager {
+
+    private var routeDemonstrateSimulator: RouteDemonstrateSimulator? = null
+
+    override var speedMultiplier: MutableLiveData<Float> = object : MutableLiveData<Float>(DEFAULT_SPEED) {
+        override fun setValue(value: Float) {
+            if (value != this.value) {
+                super.setValue(value)
+                routeDemonstrateSimulator?.setSpeedMultiplier(value)
```

Comment:
sme sa bavili, ze namiesto tohoto by malo byt `observe` ;)
uz sme to vsade zmenili na observe

Reply:
nebylo to jen pokud se nepracovalo s tou value? tu je check na `value != this.value` a tak jsi to použil i při té úpravě v SearchToolbarViewModel-u, či? ;)

Reply:
ale klidně to předělám.

Reply:
Ako to myslis? Vsade som odstranil vsetko, okrem checku `value != this.value`, lebo na tych miestach je kriticke, aby sa neposielalo to iste viackrat (zacyklilo by sa). Ale nic okrem checku v tej overridnutej metode nie je. 

Nie vsade je kriticke aby sa checkovala zmena hodnoty, niekedy aj mozes chciet, aby sa preposlala aj rovnaka viackrat. Ak ale u nas je to uz na viacerych miestach pozadovane, pohravam sa s myslienkou urobit extendnutu triedu `UniqueMutableLiveData` alebo daco take...

Reply:
vytvořil jsem ;)